pygtrie.py

```diff
@@ -1220,7 +1220,14 @@ def __init__(self, *args, **kwargs):
                 named argument is not specified on the function's prototype
                 because of Python's limitations.
         """
-        self._separator = kwargs.pop('separator', '/')
+        separator = kwargs.pop('separator', '/')
+        if not isinstance(separator, str):
```

Comment:
Tanks for your insights, I finally took the code sample you suggested to check the type